### PR TITLE
GEODE-7864: Remove null checks that are not required.(Part 2)

### PR DESCRIPTION
**Pull request by nabarunnag**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java [hunk 1]
```diff
@@ -2417,9 +2417,9 @@ public int hashCode() {
     final int prime = 31;
     int result = 1;
     result = prime * result + (checkPortability ? 1231 : 1237);
-    result = prime * result + ((classPatterns == null) ? 0 : classPatterns.hashCode());
-    result = prime * result + ((excludePatterns == null) ? 0 : excludePatterns.hashCode());
-    result = prime * result + ((identityPatterns == null) ? 0 : identityPatterns.hashCode());
+    result = prime * result + classPatterns.hashCode();
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java]: Is this supposed to be "classPatterns.uisEmpty() ? 0 : classPatterns.hashCode()" 
Anyways the hashCode for empty set returns 0. Not sure if there if there are any performance benefits.

File: geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java [hunk 2]
```diff
@@ -2434,20 +2434,11 @@ public boolean equals(Object obj) {
     AutoSerializableManager other = (AutoSerializableManager) obj;
     if (checkPortability != other.checkPortability)
       return false;
-    if (classPatterns == null) {
-      if (other.classPatterns != null)
-        return false;
-    } else if (!classPatterns.equals(other.classPatterns))
+    if (!classPatterns.equals(other.classPatterns))
```
Comment (agingade) [inline geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java]: Can be made simple by having OR condition between all 3 checks. Its up to you what style you want to follow.

File: geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java (hunk 1 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java:2420]: I went with thinking that hashCode implementation of HashSet is implemented by JDK and hence it must be designed in the already optimized state.

Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java:2420]: Its also autogenerated code via the IDE

File: geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java (hunk 2 above)
Comment (nabarunnag) [reply geode-core/src/main/java/org/apache/geode/pdx/internal/AutoSerializableManager.java:2437]: Fixed

**Pull request merged**